Resending email invitations implementation

zeynepaki opened the pull request:
* Fixes #191

zeynepaki committed:
new email method added to the email api for reminder emails

zeynepaki committed:
added the button to send reminder emails

zeynepaki committed:
Implemented the send_reminder-email method

zeynepaki committed:
Added reminder to the email being sent

zeynepaki committed:
Added new tests to test the send reminder button

alisonrclarke committed:
Merge branch 'master' into resending-email-invitations

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -310,6 +311,43 @@ def send_cookie_email(submission_participant,
+    )
+    coordinator = User.query.get(hepsubmission.coordinator)
+
+    reviewers_notified = submission_participant.user_account is not None
```
Not sure about this.  I would just set `reviewers_notified = True` to get the correct text in `submission_status.html`.

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -310,6 +311,43 @@ def send_cookie_email(submission_participant,
+                                   actor=coordinator.email,
+                                   article=record_information['recid'],
+                                   message=None,
+                                   invite_token=submission_participant.invitation_cookie,
```
The `invite_token` should be set as in `send_notification_email`, i.e.
```python
invite_token = None
if not submission_participant.user_account:
    invite_token = submission_participant.invitation_cookie
```
This ensures that the link to assign permissions is not displayed in `submission_status.html` if the permissions have already been claimed.

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -310,6 +311,43 @@ def send_cookie_email(submission_participant,
+    if not reviewers_notified:
+        message_subject = '[HEPData] Reminder to review submission {0}'.format(record_information['recid'])
+    else:
+        message_subject = '[HEPData] Reminder about submission {0}'.format(record_information['recid'])
```
Instead of using `reviewers_notified` in these conditions I would check for `if submission_participant.role == 'reviewer'` to use the first subject, `else` use the second subject.

GraemeWatt reviewed (changes requested):
Thanks, this looks great!  I've tested and it works well, but I think some small changes are needed in the new `send_reminder_email` function (commented separately).  Also, it wasn't mentioned in the original issue #191, but I made some comments elsewhere (linked from #191) that I'd like an option for the Coordinator to add a personalised message to the reminder.  Sorry that I didn't make that clear.  Could you please add a text box (similar to the one available from the "Notify Participants" button) where the Coordinator could add an optional `message` to be passed to `submission_status.html`?

ItIsJordan committed:
Merge branch 'main' into resending-email-invitations

ItIsJordan committed:
Update deprecated test code

Updates old test code using deprecated find_element_by functions in test_dashboard to use new find_element function.

ItIsJordan committed:
Allow manage_participant_status to allow POST

Updates manage_participant_status view function (permissions/manage/) to allow POST requests. This is changed to allow reminder message passing.

ItIsJordan committed:
Update send_reminder_email in email/api.py to allow message

Adds a 'message' argument to send_reminder_email to allow passing of custom message to reminder email template (which already supports this). Adds commenting.

ItIsJordan committed:
Add reminder message box to Manage Submission widget email reminder action

Adds a message box to the "Manage Submission" dashboard widget to allow writing a custom message in the reminder. Updates the form request to a POST request including any message contents.

ItIsJordan committed:
Add css for review message reminder container

Add css for review message reminder container in 'Manage Submission' dashboard widget to position text left and size textbox based on width.

ItIsJordan committed:
Update manage_participant_status (permissions/manage) to receive review message

Updates manage_participant_status (permissions/manage) view to receive review message content, and passes it to the send_reminder_email function.

ItIsJordan commented:
Following the previous discussions, I have implemented a message box in the dashboard's `Manage Submission` widget to allow a custom reminder message to be sent alongside the default message.

![dashboard_reminder](https://github.com/user-attachments/assets/7e25be38-3f03-4c37-80d9-4194d6f141c7)

Let me know if you have any feedback on styling or language.

ItIsJordan committed:
Improve message subject condition in send_reminder_email

Updates the logic in send_reminder_email to use proper variable (submission_participant.role) to determine action based on role. Updated for clarity.

ItIsJordan committed:
Remove send_reminder_email function from email/api

Removes the send_reminder_email function from emails/api as the functionality is essentially duplicated in the send_cookie_email function.

Also replaces references from send_reminder_email->send_cookie_email.

ItIsJordan committed:
Add comment to send_cookie_email

Add docstring comment to send_cookie_email

ItIsJordan committed:
Add reminder text to invite email

Adds an optional "reminder" text to the subject of the invite email in email/invite.html

ItIsJordan committed:
Remove reminder setting from send_notification_email

Alters send_notification_email to remove default setting of reminder to False when calling for the template.

ItIsJordan committed:
Remove reminder message from submission_status.html

Removes the if statement regarding adding the text "Reminder:" to the email subject in the submission status email template.

ItIsJordan committed:
Add reminder argument to send_cookie_email

Adds an optional argument to send_cookie_email to allow setting whether reminder text is displayed in html template or not

ItIsJordan committed:
Update send_cookie_email use in manage_participant_status

Update send_cookie_email use in manage_participant_status to set the reminder=True to show reminder text to user.

ItIsJordan committed:
add email_test.py

Adds new email_test.py file for testing specific email function tasks. Also adds first test function for the send_cookie_email function.

ItIsJordan committed:
Fix test folder directory in test_send_cookie_email

ItIsJordan committed:
Merge branch 'main' into resending-email-invitations

GraemeWatt reviewed (approved):
This looks great!  Thanks to @zeynepaki for starting this task and to @ItIsJordan for completing it.  I'll merge now.
